src/config/loading.rs

```diff
@@ -35,6 +37,121 @@ lazy_static! {
     pub static ref CONFIG_PATHS: Mutex<Vec<ConfigPath>> = Mutex::default();
 }
 
+fn component_name(path: &Path) -> Result<String, Vec<String>> {
+    path.file_stem()
+        .and_then(|name| name.to_str())
+        .map(|name| name.to_string())
+        .ok_or_else(|| vec![format!("Couldn't get component name for file: {:?}", path)])
+}
+
+trait ConfigLoader: Sized + serde::de::DeserializeOwned {
```

Comment:
Maybe a better name for this would be `LoadableConfig` or similar? `ConfigLoader` implies it's a separate thing that loads config, but these are all config objects themselves that can be loaded.

Reply:
Thank you, I still have some issues with this kind of concept 👍